Logger/LogFactory.cs

```diff
@@ -1,10 +1,19 @@
-﻿namespace Logger;
+﻿using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+
+namespace Logger;
 
 public class LogFactory
 {
-    public BaseLogger CreateLogger(string className)
+    private string? _filePath;
+    public BaseLogger? CreateLogger(string className)
     {
+        return _filePath == null ? null : (BaseLogger)new FileLogger(className);
```

Comment:
Why is this cast needed?
```suggestion
        return _filePath == null ? null : new FileLogger(className);
```